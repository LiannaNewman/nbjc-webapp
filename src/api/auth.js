import { useAuth0 } from '@auth0/auth0-react';

const getToken = async () => {
  const { getAccessTokenSilently } = useAuth0();
  const token = await getAccessTokenSilently();

  return token;
};

const getUserID = async () => {
  const { user } = useAuth0();
  const userID = user.sub;
  return userID;
};

export { getToken, getUserID };
