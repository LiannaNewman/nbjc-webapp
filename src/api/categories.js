import fetch from 'node-fetch';

/**
 * Get categories by search term
 * @param {Object} searchOpts
 * @param {string} searchOpts.searchTerm - name of space to search
 * @returns {Promise} - resolves to list of businesses or errors when none found
*/
const getCategories = async (searchOpts) => {
  const url = new URL(process.env.REACT_APP_API_HOST);
  url.pathname = '/api/v1/categories';
  url.searchParams.append('search', searchOpts.searchTerm);
  url.searchParams.append('page', 1);
  url.searchParams.append('per_page', 5);
  const results = await fetch(url.href);
  // todo: add error handling
  const { data } = await results.json();
  const formattedCategories = data.map((category) => ({
    title: category.title,
    alias: category.alias,
    value: `${category.id}`,
  }));
  return new Promise((resolve) => {
    resolve(formattedCategories);
  });
};

export default getCategories;
