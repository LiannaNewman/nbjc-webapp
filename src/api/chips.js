export default [{
  name: 'Accessible',
  value: '1',
}, {
  name: 'ATM Available',
  value: '2',
}, {
  name: 'ASL Spoken',
  value: '3',
}, {
  name: 'Black Friendly',
  value: '4',
}, {
  name: 'Black Owned',
  value: '5',
}, {
  name: 'Gender-Neutral Restroom',
  value: '6',
}, {
  name: 'Inclusive',
  value: '7',
}, {
  name: 'LGBTQIA+ Friendly',
  value: '8',
}, {
  name: 'Multilingual',
  value: '9',
}, {
  name: 'Person of Color owned',
  value: '10',
}, {
  name: 'Price Conscious',
  value: '11',
}, {
  name: 'Trans-Trained',
  value: '12',
}, {
  name: 'Wheelchair Ramp',
  value: '13',
}];
