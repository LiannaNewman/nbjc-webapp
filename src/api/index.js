import getAllIndicators from './indicators';
import getSearchResults from './search';

export {
  getSearchResults,
  getAllIndicators,
};
