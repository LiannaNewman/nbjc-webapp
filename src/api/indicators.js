import fetch from 'node-fetch';

const getAllIndicators = async () => {
  const url = new URL(process.env.REACT_APP_API_HOST);
  url.pathname = '/api/v1/indicators';
  const results = await fetch(url.href);
  // todo: add error handling
  const { data } = await results.json();
  const formattedIndicators = data.map((indicator) => ({
    name: indicator.name,
    value: `${indicator.id}`,
  }));
  return new Promise((resolve) => {
    resolve(formattedIndicators);
  });
};

export default getAllIndicators;
