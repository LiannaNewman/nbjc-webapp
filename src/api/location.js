import fetch from 'node-fetch';

/**
 * Get a location based on lat and lng
 * @param {Object} locationParams
 * @param {number} locationParams.lat - latitude to look up
 * @param {number} locationParams.lng - longitude to look up
 * @returns {Promise} - resolves to a reverse geocoded lookup object
*/
const getLocation = async (locationParams) => {
  const url = new URL(process.env.REACT_APP_API_HOST);
  url.pathname = '/api/v1/geolocations';
  url.searchParams.append('lat', locationParams.lat);
  url.searchParams.append('lng', locationParams.lng);
  const result = await fetch(url.href);
  return result.json();
};

export default getLocation;
