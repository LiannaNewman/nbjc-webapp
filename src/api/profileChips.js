import fetch from 'node-fetch';

const getProfileChips = async () => {
  const url = '/api/v1/identities';

  const results = await fetch(url, {
    method: 'GET',
    mode: 'cors',
    headers: {
      'Content-Type': 'application/json',
      Accept: 'application/json',
    },
  });
  /* eslint-disable-next-line */
  return results;
};

export default getProfileChips;
