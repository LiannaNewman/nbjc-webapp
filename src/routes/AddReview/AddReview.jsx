import React, { useState, useEffect, useContext } from 'react';
import { useParams } from 'react-router-dom';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';

import Button from '@material-ui/core/Button';
import CircularProgress from '@material-ui/core/CircularProgress';
import Typography from '@material-ui/core/Typography';

import {
  getReviewForSpaceAndUser,
  getSpace,
  postReview,
} from '../../api';
import Review from '../../components/AddSpacePage/Review';
import Success from '../../components/AddSpacePage/Success';
import ErrorSnackbar from '../../components/ErrorSnackbar';
import { UserContext } from '../../context/UserContext';

import withUser from '../AuthenticatedRoute';

const styles = (theme) => ({
  root: {
    [theme.breakpoints.up('xs')]: {
      margin: '0 20px',
    },
    [theme.breakpoints.up('mobile')]: {
      margin: '0 100px',
    },
  },
  reviewSubtitle: {
    [theme.breakpoints.up('xs')]: {
      marginBottom: 60,
    },
    [theme.breakpoints.up('mobile')]: {
      marginBottom: 40,
    },
  },
  reviewFooter: {
    display: 'flex',
    justifyContent: 'center',
    alignItems: 'center',
  },
  reviewSubmit: {
    [theme.breakpoints.up('mobile')]: {
      width: 254,
    },
  },
  successFooter: {
    display: 'block !important',
  },
  successButton: {
    [theme.breakpoints.up('mobile')]: {
      width: 254,
    },
    margin: '0 auto',
    display: 'block',
    marginBottom: 40,
  },
});

const AddReview = ({ classes }) => {
  const { spaceId } = useParams();
  const [space, setSpace] = useState(null);
  const [pageStatus, setPageStatus] = useState('review');
  const [reviewData, setReviewData] = useState();
  const [snackbarOpen, setSnackbarOpen] = useState(false);
  const { user } = useContext(UserContext);

  useEffect(() => {
    async function fetchData() {
      const intId = parseInt(spaceId, 10);
      const [spaceData, userReview] = await Promise.all([
        getSpace(intId),
        getReviewForSpaceAndUser({
          spaceId: intId,
          userId: user.userId,
        }),
      ]);
      if (userReview.data.exists) {
        setSpace(spaceData.data);
        setPageStatus('reviewExists');
      } else {
        setSpace(spaceData.data);
      }
    }
    try {
      fetchData();
      setPageStatus('review');
    } catch (e) {
      setSnackbarOpen(true);
    }
  }, [spaceId]);

  useEffect(() => {
    async function postReviewData(data) {
      await postReview(data);
      setPageStatus('success');
    }
    if (pageStatus === 'reviewSubmitted') {
      try {
        postReviewData({
          spaceId,
          rating: reviewData.rating,
          detail: reviewData.review,
          anonymous: reviewData.anon,
          ...user,
        });
      } catch (e) {
        setPageStatus('review');
        setSnackbarOpen(true);
      }
    }
  });

  const saveReview = (formData) => {
    setReviewData(formData);
    setPageStatus('reviewSubmitted');
  };
  const loading = !space || pageStatus === 'reviewSubmitted';

  return (
    <div className={classes.root}>
      {space && pageStatus === 'review' && (
        <Review
          title={(isDesktop) => (
            <>
              <Typography variant={isDesktop ? 'h2' : 'h4'} align="center">
                Rate and Review
              </Typography>
              <Typography
                variant={isDesktop ? 'h4' : 'subtitle1'}
                align="center"
                className={classes.reviewSubtitle}
              >
                {`You can rate ${space.name} anonymously.`}
              </Typography>
            </>
          )}
          showBack={false}
          submitLabel="Submit"
          onNext={saveReview}
          disableSubmit={() => loading}
          overrideClasses={{
            footer: classes.reviewFooter,
            submitButton: classes.reviewSubmit,
          }}
        />
      )}
      {space && pageStatus === 'success' && (
        <Success
          title="Your Review Was Submitted!"
          subtitle={`Thank you for rating and reviewing ${space.name} on Lavender Book.`}
          primaryButton={(isDesktop) => (
            <Button
              variant="contained"
              align="center"
              fullWidth={!isDesktop}
              href={`/spaces/${space.id}`}
              color="primary"
              disableElevation
              className={classes.successButton}
            >
              Go to space
            </Button>
          )}
          overrideClasses={{
            buttonWrapper: classes.successFooter,
            secondaryButton: classes.successButton,
          }}
        />
      )}
      {space && pageStatus === 'reviewExists' && (
        <Success
          title="Your Review exists for this space"
          subtitle="You have already submitted your review for this space"
          primaryButton={(isDesktop) => (
            <Button
              variant="contained"
              align="center"
              fullWidth={!isDesktop}
              href={`/spaces/${space.id}`}
              color="primary"
              disableElevation
              className={classes.successButton}
            >
              Go to space
            </Button>
          )}
          overrideClasses={{
            buttonWrapper: classes.successFooter,
            secondaryButton: classes.successButton,
          }}
        />
      )}
      {loading && <CircularProgress color="secondary" />}
      <ErrorSnackbar
        snackbarOpen={snackbarOpen}
        onClose={() => setSnackbarOpen(false)}
        body="There was an error posting the reivew. Please try again later"
      />
    </div>
  );
};

AddReview.propTypes = {
  classes: PropTypes.shape({}).isRequired,
};

export default withUser(withStyles(styles)(AddReview));
