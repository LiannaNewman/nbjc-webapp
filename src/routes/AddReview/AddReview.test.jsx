import React from 'react';
import {
  render,
  waitFor,
  fireEvent,
  screen,
} from '@testing-library/react';

import AddReview from './AddReview';
import { getSpace, postReview, getReviewForSpaceAndUser } from '../../api';

jest.mock('react-router-dom', () => ({
  useParams: () => ({ spaceId: 1 }),
}));

jest.mock('react-promise-tracker', () => ({
  trackPromise: jest.fn(),
  usePromiseTracker: () => ({ promiseInProgress: false }),
}));

jest.mock('../../api', () => ({
  getSpace: jest.fn(),
  postReview: jest.fn(),
  getReviewForSpaceAndUser: jest.fn(),
}));

describe('AddReview', () => {
  beforeEach(() => {
    getSpace.mockImplementation(() => Promise.resolve({
      data: {
        name: 'abc',
      },
    }));

    getReviewForSpaceAndUser.mockImplementation(() => Promise.resolve({
      data: {
        exists: false,
      },
    }));

    postReview.mockImplementation(() => Promise.resolve({}));
  });

  it('should show the form when review is not present', async () => {
    const { asFragment, getByTestId } = render(<AddReview />);
    const form = await waitFor(() => getByTestId('review-form'));
    expect(form).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show already submitted when review is present', async () => {
    getReviewForSpaceAndUser.mockImplementation(() => Promise.resolve({
      data: {
        exists: true,
      },
    }));
    const { asFragment, getByText } = render(<AddReview />);
    const cta = await waitFor(() => getByText('Your Review exists for this space'));
    expect(cta).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });

  it('should show success page when a review is submitted', async () => {
    const {
      getByTestId,
      getByText,
      asFragment,
    } = render(<AddReview />);
    const form = await waitFor(() => getByTestId('review-form'));
    expect(form).toBeDefined();
    const starRating = screen.getByTestId('star-rating-unfilled-3');
    fireEvent.click(starRating);
    const review = screen.getByTestId('review-content').querySelector('textarea');
    fireEvent.change(review, { target: { value: 'This is a review' } });
    const submit = screen.getByText('Submit').closest('button');
    fireEvent.click(submit);
    expect(postReview).toHaveBeenCalled();
    const success = await waitFor(() => getByText('Your Review Was Submitted!'));
    expect(success).toBeDefined();
    expect(asFragment()).toMatchSnapshot();
  });
});
