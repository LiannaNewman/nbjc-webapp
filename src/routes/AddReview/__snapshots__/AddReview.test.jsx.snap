// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddReview should show already submitted when review is present 1`] = `
<DocumentFragment>
  <div
    class="AddReview-root-27"
  >
    <h4
      class="MuiTypography-root Success-pageTitle-53 MuiTypography-h4 MuiTypography-alignCenter"
    >
      Your Review exists for this space
    </h4>
    <div
      class="Success-check-54"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Success-check-54 MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
    <h6
      class="MuiTypography-root Success-body-55 MuiTypography-subtitle1 MuiTypography-alignCenter"
    >
      You have already submitted your review for this space
    </h6>
    <div
      class="Success-buttonWrapper-56 AddReview-successFooter-31"
    >
      <a
        align="center"
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained AddReview-successButton-32 MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
        href="/spaces/undefined"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Go to space
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        align="center"
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined AddReview-successButton-32 MuiButton-outlinedSecondary MuiButton-disableElevation MuiButton-fullWidth"
        href="/search"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Search for a space
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AddReview should show success page when a review is submitted 1`] = `
<DocumentFragment>
  <div
    class="AddReview-root-57"
  >
    <h4
      class="MuiTypography-root Success-pageTitle-83 MuiTypography-h4 MuiTypography-alignCenter"
    >
      Your Review Was Submitted!
    </h4>
    <div
      class="Success-check-84"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root Success-check-84 MuiSvgIcon-colorPrimary"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M12 2C6.48 2 2 6.48 2 12s4.48 10 10 10 10-4.48 10-10S17.52 2 12 2zm-2 15l-5-5 1.41-1.41L10 14.17l7.59-7.59L19 8l-9 9z"
        />
      </svg>
    </div>
    <h6
      class="MuiTypography-root Success-body-85 MuiTypography-subtitle1 MuiTypography-alignCenter"
    >
      Thank you for rating and reviewing abc on Lavender Book.
    </h6>
    <div
      class="Success-buttonWrapper-86 AddReview-successFooter-61"
    >
      <a
        align="center"
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-contained AddReview-successButton-62 MuiButton-containedPrimary MuiButton-disableElevation MuiButton-fullWidth"
        href="/spaces/undefined"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Go to space
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
      <a
        align="center"
        aria-disabled="false"
        class="MuiButtonBase-root MuiButton-root MuiButton-outlined AddReview-successButton-62 MuiButton-outlinedSecondary MuiButton-disableElevation MuiButton-fullWidth"
        href="/search"
        tabindex="0"
      >
        <span
          class="MuiButton-label"
        >
          Search for a space
        </span>
        <span
          class="MuiTouchRipple-root"
        />
      </a>
    </div>
  </div>
</DocumentFragment>
`;

exports[`AddReview should show the form when review is not present 1`] = `
<DocumentFragment>
  <div
    class="AddReview-root-1"
  >
    <form
      data-testid="review-form"
    >
      <h4
        class="MuiTypography-root MuiTypography-h4 MuiTypography-alignCenter"
      >
        Rate and Review
      </h4>
      <h6
        class="MuiTypography-root AddReview-reviewSubtitle-2 MuiTypography-subtitle1 MuiTypography-alignCenter"
      >
        You can rate abc anonymously.
      </h6>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Rate this space
      </h6>
      <div>
        <button
          class="MuiButtonBase-root MuiIconButton-root StarRating-button-17 MuiIconButton-colorPrimary"
          data-testid="star-rating-unfilled-1"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root StarRating-button-17 MuiIconButton-colorPrimary"
          data-testid="star-rating-unfilled-2"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root StarRating-button-17 MuiIconButton-colorPrimary"
          data-testid="star-rating-unfilled-3"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root StarRating-button-17 MuiIconButton-colorPrimary"
          data-testid="star-rating-unfilled-4"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
        </button>
        <button
          class="MuiButtonBase-root MuiIconButton-root StarRating-button-17 MuiIconButton-colorPrimary"
          data-testid="star-rating-unfilled-5"
          tabindex="0"
          type="button"
        >
          <span
            class="MuiIconButton-label"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M22 9.24l-7.19-.62L12 2 9.19 8.63 2 9.24l5.46 4.73L5.82 21 12 17.27 18.18 21l-1.63-7.03L22 9.24zM12 15.4l-3.76 2.27 1-4.28-3.32-2.88 4.38-.38L12 6.1l1.71 4.04 4.38.38-3.32 2.88 1 4.28L12 15.4z"
              />
            </svg>
          </span>
        </button>
        <span
          class="MuiTypography-root Review-ratingText-11 MuiTypography-caption"
        >
          0 stars
        </span>
      </div>
      <h6
        class="MuiTypography-root MuiTypography-h6"
      >
        Share your thoughts
      </h6>
      <div
        class="MuiFormControl-root MuiTextField-root Review-review-12 MuiFormControl-fullWidth"
        data-testid="review-content"
      >
        <label
          class="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
          data-shrink="false"
        >
          Write a Review
        </label>
        <div
          class="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-multiline MuiOutlinedInput-multiline"
        >
          <textarea
            aria-invalid="false"
            class="MuiInputBase-input MuiOutlinedInput-input MuiInputBase-inputMultiline MuiOutlinedInput-inputMultiline"
            name="review"
            rows="5"
          />
          <fieldset
            aria-hidden="true"
            class="PrivateNotchedOutline-root-19 MuiOutlinedInput-notchedOutline"
          >
            <legend
              class="PrivateNotchedOutline-legendLabelled-21"
            >
              <span>
                Write a Review
              </span>
            </legend>
          </fieldset>
        </div>
        <p
          class="MuiFormHelperText-root MuiFormHelperText-contained"
        >
          <span
            class="MuiTypography-root MuiTypography-caption MuiTypography-colorPrimary"
          >
            Required
          </span>
          <span
            class="MuiTypography-root Review-remaining-13 MuiTypography-caption"
          >
            500 characters remaining
          </span>
        </p>
      </div>
      <label
        class="MuiFormControlLabel-root Review-checkbox-14"
      >
        <span
          aria-disabled="false"
          class="MuiButtonBase-root MuiIconButton-root PrivateSwitchBase-root-23 MuiCheckbox-root MuiCheckbox-colorPrimary MuiIconButton-colorPrimary"
        >
          <span
            class="MuiIconButton-label"
          >
            <input
              class="PrivateSwitchBase-input-26"
              data-indeterminate="false"
              name="anon"
              type="checkbox"
              value=""
            />
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M19 5v14H5V5h14m0-2H5c-1.1 0-2 .9-2 2v14c0 1.1.9 2 2 2h14c1.1 0 2-.9 2-2V5c0-1.1-.9-2-2-2z"
              />
            </svg>
          </span>
          <span
            class="MuiTouchRipple-root"
          />
        </span>
        <span
          class="MuiTypography-root MuiFormControlLabel-label MuiTypography-body1"
        >
          <p
            class="MuiTypography-root MuiTypography-body2"
          >
            Publish my rating and review anonymously.
          </p>
        </span>
      </label>
      <div
        class="Review-footer-15 AddReview-reviewFooter-3"
      >
        <button
          class="MuiButtonBase-root MuiButton-root MuiButton-contained Review-submitButton-16 AddReview-reviewSubmit-4 MuiButton-containedSecondary MuiButton-disableElevation Mui-disabled MuiButton-fullWidth Mui-disabled"
          disabled=""
          tabindex="-1"
          type="submit"
        >
          <span
            class="MuiButton-label"
          >
            Submit
          </span>
        </button>
      </div>
    </form>
  </div>
</DocumentFragment>
`;
